### ISSUE TITLE
[SCREENREADER]: Legends must be first child element of fieldesets for tabbing navigation

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#767]
## Issue

Our forms generator is wrapping the `<legend>` in a `<div>` in the radio and checkbox components. This is causing screen readers to not announce the legend text when device users are navigating through form components with the `TAB` key. I've attached some code showing the issue and a screenshot below.

## Point of Contact

Trevor

## Acceptance Criteria

* As a screen reader user, I want to hear the `<legend>` read out when I navigate to a first radio button or checkbox. This should happen if I am navigating using the `TAB` key or the virtual cursor.

## Environment
* Windows 10, MacOS
* Chrome, Firefox, IE11, Safari
* JAWS, NVDA, VoiceOver

## Possible Fixes (optional)
Making the `<legend>` the first child of the `<fieldset>` restored the functionality on all screen reader combinations I tried.

```html
<fieldset>
  <legend>This legend is unwrapped and should be read</legend>
  <div>
    <label>
      <input checked name="unwrapped-radio" type="radio" value="first unwrapped">
      First unwrapped radio input
    </label>
    <label>
      <input name="unwrapped-radio" type="radio" value="second unwrapped">
      Second unwrapped radio input
    </label>
  </div>
</fieldset>
```

## WCAG or Vendor Guidance (optional)
* [Name, Role, Value: Understanding SC 4.1.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/ensure-compat-rsv.html)

## Screenshots or Trace Logs

![Screen Shot 2019-07-31 at 3 34 58 PM](https://user-images.githubusercontent.com/934879/62246593-914b8000-b3a9-11e9-8b41-4c8386b6361d.png)

**Comment by rianfowler** [comment]
[Example](https://design.va.gov/components/form-controls#checkboxes)

**Comment by rianfowler** [comment]
- review everywhere we're using legend and make sure it's the first child in the `<fieldset>`

**Comment by bkjohnson** [comment]
I've confirmed this markup structure happening on the _Demographic Information_ part of the ["Apply for health care" form](https://staging.va.gov/health-care/apply/application/veteran-information/demographic-information).


![image](https://user-images.githubusercontent.com/2008881/86045506-94f3e680-ba00-11ea-8870-4344f48d44b4.png)

The rendered page:

![image](https://user-images.githubusercontent.com/2008881/86045682-ca003900-ba00-11ea-80f6-ea33bded4b58.png)

**Comment by bkjohnson** [comment]
This particular problem (and likely others) can be traced to the `<ObjectField>` component being used with top level `<div>` tags inside of a `<fieldset>`:

https://github.com/department-of-veterans-affairs/vets-website/blob/20a303ad06309ccf5f41022c54d341cc60671edc/src/platform/forms-system/src/js/fields/ObjectField.jsx#L228-L231

https://github.com/department-of-veterans-affairs/vets-website/blob/20a303ad06309ccf5f41022c54d341cc60671edc/src/platform/forms-system/src/js/fields/ObjectField.jsx#L293-L295

**Comment by meganhkelley** [comment]
@bkjohnson was this completed? If not, can you provide an update on what work is remaining? Thanks!

**Comment by timwright12** [comment]
@meganhkelley @bkjohnson Is this still being worked on? If so, I'd like to follow along to help familiarize myself with some og the system and processes (if there are no objections)

**Comment by bkjohnson** [comment]
I've merged a PR which should address this, and we can verify it in staging. If the issue is resolved we will create a follow up PR to make the change available in production.

**Comment by k80bowman** [comment]
@bkjohnson did you ever verify that your fix works in staging?

**Comment by bkjohnson** [comment]
No, I was going to let an accessibility expert verify it since I can't test with screen readers for Windows or MacOS.

**Comment by k80bowman** [comment]
That sounds fine, do you know if there are good pages to test this out on?

**Comment by bkjohnson** [comment]
In my original testing I used the "Demographic Information" page on the [Apply for health care form](https://staging.va.gov/health-care/apply/application/veteran-information/demographic-information).

**Comment by k80bowman** [comment]
@CourtneyChambers77 could you please verify that Brooks's solution is working? Please read through the issue above and if you have questions about it, let me know. You can test it using the link Brooks provided above.

**Comment by CourtneyChambers77** [comment]
@bkjohnson  
the tabbing order is good able to focus and hear the elements with SR
